arelle/RuntimeOptions.py

```diff
@@ -147,7 +147,12 @@ def __repr__(self) -> str:
         )
         return f"{self.__class__.__name__}({r})"
 
-    def __post_init__(self, pluginOptions: Optional[dict[str, RuntimeOptionValue]]) -> None:
+     def ensure_value(self, attr, value):
+        if not hasattr(self, attr) or getattr(self, attr) is None:
+            setattr(self, attr, value)
+        return getattr(self, attr)
+    
+     def __post_init__(self, pluginOptions: Optional[dict[str, RuntimeOptionValue]]) -> None:
```

Comment:
The indention is slightly off:
```suggestion
    def ensure_value(self, attr, value):
        if not hasattr(self, attr) or getattr(self, attr) is None:
            setattr(self, attr, value)
        return getattr(self, attr)

    def __post_init__(self, pluginOptions: Optional[dict[str, RuntimeOptionValue]]) -> None:
```